File: app/src/main/java/com/flyjingfish/lightaop/DefaultLightAop.java
package com.flyjingfish.lightaop;

import android.util.Log;

import org.aspectj.lang.ProceedingJoinPoint;

public class DefaultLightAop implements BaseLightAop{
    @Override
    public void beforeInvoke(DebugLog debugLog) {
        Log.e("DefaultLightAop","=====beforeInvoke====");
    }

    @Override
    public Object invoke(ProceedingJoinPoint joinPoint,DebugLog debugLog) {
        Log.e("DefaultLightAop","=====invoke====");
        Object oj = null;
        try {
            oj= joinPoint.proceed();
        } catch (Throwable e) {
            throw new RuntimeException(e);
        }
        return oj;
    }

    @Override
    public void afterInvoke(DebugLog debugLog) {
        Log.e("DefaultLightAop","=====afterInvoke====");
    }
}

File: light-aop-core/src/main/java/com/flyjingfish/light_aop_core/aop/DefaultLightAop.kt
package com.flyjingfish.light_aop_core.aop

import android.util.Log
import com.flyjingfish.light_aop_annotation.BaseLightAop
import com.flyjingfish.light_aop_core.annotations.DefaultAnnotation
import org.aspectj.lang.ProceedingJoinPoint

class DefaultLightAop : BaseLightAop<DefaultAnnotation?> {
    override fun beforeInvoke(defaultAnnotation: DefaultAnnotation?) {
        Log.e("DefaultLightAop", "=====beforeInvoke====")
    }

    override operator fun invoke(
        joinPoint: ProceedingJoinPoint?,
        defaultAnnotation: DefaultAnnotation?
    ): Any? {
        Log.e("DefaultLightAop", "=====invoke====")
        var oj: Any? = null
        oj = try {
            joinPoint!!.proceed()
        } catch (e: Throwable) {
            throw RuntimeException(e)
        }
        return oj
    }

    override fun afterInvoke(defaultAnnotation: DefaultAnnotation?) {
        Log.e("DefaultLightAop", "=====afterInvoke====")
    }
}